pkg/runctl/controller.go

```diff
@@ -279,6 +288,17 @@ func (c *Controller) syncHandler(key string) error {
 	}
 
 	if pipelineRun.GetStatus().State == api.StateUndefined {
+		maintenanceMode, err := c.isMaintenanceMode()
+		if err != nil {
+			c.recorder.Event(pipelineRunAPIObj, corev1.EventTypeNormal, api.EventReasonMaintenanceMode, err.Error())
```

Comment:
I think we must use a different reason here, because operators must react on it (alerting). This is similar to [loading the pipeline run config](https://github.com/SAP/stewardci-core/blob/475c7fc246d84193d280a7afd409f0fe2017f894/pkg/runctl/controller.go#L290-L295). Maybe we should a reason to indicate internal system errors.

Reply:
I think this fits best: 
6aa1d34

Reply:
`RecoverableError` doesn't really express that the system's configuration is broken and must be fixed. Even worse, it suggest that the error can be recovered automatically.

I propose `ControllerConfigError`.